Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -105,71 +101,10 @@ public void CreateRoslynAnalyzerProvider_NullServer_ThrowsArgumentNullException(
-            AuthorizationHeader("da39a3ee5e6b4b0d3255bfef95601890afd80709", null).Should().Be("Basic ZGEzOWEzZWU1ZTZiNGIwZDMyNTViZmVmOTU2MDE4OTBhZmQ4MDcwOTo=");
-            AuthorizationHeader("da39a3ee5e6b4b0d3255bfef95601890afd80709", string.Empty).Should().Be("Basic ZGEzOWEzZWU1ZTZiNGIwZDMyNTViZmVmOTU2MDE4OTBhZmQ4MDcwOTo=");
-            AuthorizationHeader("admin", "password").Should().Be("Basic YWRtaW46cGFzc3dvcmQ=");
-
-            static string AuthorizationHeader(string userName, string password) =>
-                GetHeader(PreprocessorObjectFactory.CreateHttpClient(userName, password, null, null), "Authorization");
-        }
-
-        [TestMethod]
-        public void CreateHttpClient_UserAgent()
-        {
-            var scannerVersion = typeof(WebClientDownloaderTest).Assembly.GetName().Version.ToDisplayString();
-            var client = PreprocessorObjectFactory.CreateHttpClient(null, null, null, null);
-            GetHeader(client, "User-Agent").Should().Be($"SonarScanner-for-.NET/{scannerVersion}");
-
-            // This asserts wrong "UserAgent" header. Should be removed as part of https://github.com/SonarSource/sonar-scanner-msbuild/issues/1421
-            GetHeader(client, "UserAgent").Should().Be($"ScannerMSBuild/{scannerVersion}");
-        }
-
-        [TestMethod]
-        public void CreateHttpClient_SemicolonInUsername()
-        {
-            Action act = () => PreprocessorObjectFactory.CreateHttpClient("user:name", null, null, null);
-            act.Should().ThrowExactly<ArgumentException>().WithMessage("username cannot contain the ':' character due to basic authentication limitations");
-        }
-
-        [TestMethod]
-        public void CreateHttpClient_AccentsInUsername()
-        {
-            Action act = () => PreprocessorObjectFactory.CreateHttpClient("héhé", "password", null, null);
-            act.Should().ThrowExactly<ArgumentException>().WithMessage("username and password should contain only ASCII characters due to basic authentication limitations");
-        }
-
-        [TestMethod]
-        public void CreateHttpClient_AccentsInPassword()
-        {
-            Action act = () => PreprocessorObjectFactory.CreateHttpClient("username", "héhé", null, null);
-            act.Should().ThrowExactly<ArgumentException>().WithMessage("username and password should contain only ASCII characters due to basic authentication limitations");
-        }
-
-        [TestMethod]
-        public void CreateHttpClient_UsingClientCert()
-        {
-            Action act = () => PreprocessorObjectFactory.CreateHttpClient(null, null, "certtestsonar.pem", "dummypw");
-            act.Should().NotThrow();
-        }
-
-        [TestMethod]
-        public void CreateHttpClient_MissingCert() =>
-            FluentActions.Invoking(() => PreprocessorObjectFactory.CreateHttpClient(null, null, "missingcert.pem", "dummypw")).Should().Throw<CryptographicException>();
-
```

Comment:
All those tests have been moved to [`WebClientDownloaderBuilderTest`](https://github.com/SonarSource/sonar-scanner-msbuild/pull/1505/files#diff-0743c9c00891a91c10a0015dad6cffc0b6ec285cc596387aa2dcc4f9b52d5273)